### PR TITLE
Template Part Block: Fallback to missing state if slug or theme is invalid

### PR DESCRIPTION
**Pull request by david-szabo97**
## Description
Template part block with undefined theme attribute caused the block to render itself.

## How has this been tested?
1. Create a `test` template part file in your theme (TT1 blocks) with the content below:
```html
<!-- wp:template-part {"slug":"header"} /-->
```
2. Open site editor
3. Add template part block
4. Choose existing, select the template part you just created: `test`
5. It should show the missing state

## Screenshots 
| before | after |
| - | - |
| image | image |

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by jeyip** [comment]
## Testing
### Behavior
- [x] "Test" template part shows missing template part state

### Browsers
- [x] Edge
- [x] Firefox
- [x] Safari
- [x] Chrome
- [ ] ~IE11~ (see https://github.com/WordPress/gutenberg/issues/28249)

File: packages/block-library/src/template-part/edit/index.js [hunk 1]
```diff
@@ -72,7 +72,12 @@ export default function TemplatePartEdit( {
 	const isPlaceholder = ! slug;
 	const isEntityAvailable = ! isPlaceholder && ! isMissing;
 
-	if ( ! isPlaceholder && isMissing ) {
+	// We don't want to render a missing state if we have any inner blocks.
+	// A new template part is automatically created if we have any inner blocks but no entity.
+	if (
+		innerBlocks.length === 0 &&
+		( ( slug && ! theme ) || ( slug && isMissing ) )
```
Comment (jeyip) [inline packages/block-library/src/template-part/edit/index.js:79]: > A new template part is automatically created if we have any inner blocks but no entity.

Trying to wrap my head around why this happens 🤔. The rest of the conditional makes sense to me.

**Review by mattwiebe — APPROVED** [review]
Works as advertised

File: packages/block-library/src/template-part/edit/index.js (hunk 1 above)
Comment (jeyip) [reply packages/block-library/src/template-part/edit/index.js:79]: Did some debugging and still a bit stumped by where the automatic template part creation happens. Would you mind elaborating more on this?

**Review by jeyip — COMMENTED** [review]
I left one comment. +1 from me whenever we resolve the discussion.

File: packages/block-library/src/template-part/edit/index.js (hunk 1 above)
Comment (david-szabo97) [reply packages/block-library/src/template-part/edit/index.js:79]: Don't worry, it took me a while to figure out as well :D `TemplatePartPlaceholder` component takes care of the creation.

https://github.com/WordPress/gutenberg/blob/f31c35c5756091392f64c203c1e0a57c6badab8e/packages/block-library/src/template-part/edit/placeholder/index.js#L39-L52

**Review by jeyip — APPROVED** [review]

File: packages/block-library/src/template-part/edit/index.js (hunk 1 above)
Comment (jeyip) [reply packages/block-library/src/template-part/edit/index.js:79]: The changes look good 👍

**Pull request merged**